reconcile/aus/base.py

```diff
@@ -739,6 +775,55 @@ def verify_lock_should_skip(
     return False
 
 
+def _create_upgrade_policy(
+    next_schedule: str, spec: ClusterUpgradeSpec, version: str
+) -> AbstractUpgradePolicy:
+    if spec.cluster.is_rosa_hypershift():
+        return ControlPlaneUpgradePolicy(
+            cluster=spec.cluster,
+            version=version,
+            schedule_type="manual",
+            next_run=next_schedule,
+        )
+    return ClusterUpgradePolicy(
+        cluster=spec.cluster,
+        version=version,
+        schedule_type="manual",
+        next_run=next_schedule,
+    )
+
+
+def _calulate_node_pool_diffs(
+    ocm_api: OCMBaseClient, spec: ClusterUpgradeSpec, now: datetime
+) -> Optional[UpgradePolicyHandler]:
+    node_pools = get_node_pools(ocm_api, spec.cluster.id)
+    if node_pools:
+        for pool in sorted(node_pools, key=lambda x: x["id"]):
+            current = get_node_pool_upgrade_policies(
+                ocm_api, spec.cluster.id, pool["id"]
+            )
+            if current:
+                continue
+            pool_version_id = pool.get("version", {}).get("id")
+            pool_version = get_version(ocm_api, pool_version_id)["raw_id"]
+            if semver.match(pool_version, f"<{spec.current_version}"):
+                next_schedule = verify_schedule_should_skip(spec, now)
```

Comment:
I wonder if we should really check the schedule here. this may end up having nodepools upgraded 1 day (or even more) after the control plane

Reply:
Yes, I think we should honor the maintenance window here, cause node pool upgrades might cause pod restarts. WDYT?

Reply:
I am not sure. With a weekly schedule, we'll end up having control plane upgrade one week, and 1 worker pool upgrade the following week, and repeat with the number of worker pool.

Whereas if we do not check the schedule for worker pools, we get the control plane upgrade on schedule, followed right after by each worker pool upgrades (one after the other). This looks much more like a single upgrade, on the configured schedule. WDYT?

Reply:
Agreed!